src/main/java/me/ryanhamshire/GriefPrevention/PlayerEventHandler.java

```diff
@@ -2620,6 +2636,25 @@ else if(playerData.shovelMode == ShovelMode.Subdivide)
 			}
 		}
 	}
+
+	// Stops an untrusted player from removing a book from a lectern
+	@EventHandler(priority = EventPriority.LOWEST, ignoreCancelled = true)
+	void onTakeBook(PlayerTakeLecternBookEvent event)
+	{
+		Player player = event.getPlayer();
+		PlayerData playerData = this.dataStore.getPlayerData(player.getUniqueId());
+		Claim claim = this.dataStore.getClaimAt(event.getLectern().getLocation(), false, playerData.lastClaim);
+		if (claim != null)
+		{
+			playerData.lastClaim = claim;
+			if (claim.allowContainers(player) != null)
+			{
+				event.setCancelled(true);
+				player.closeInventory();
+				GriefPrevention.sendMessage(player, TextMode.Err, claim.allowContainers(player));
```

Comment:
Store result of `claim.allowContainers(player)` and reuse, rather than running the check twice.

Reply:
You didn't change this bit.

Reply:
WTF is wrong with me today... jeez haha! Clearly I need more coffee. Ill change it right up.